### PR TITLE
Fix: variable selections should persist

### LINKED ISSUE #14747: Variable selection should persist when editing a cell
When using variables in a dashboard, the variable selection should persist when you are editing a cell. currently, the values revert to the first selection.

### PR DESCRIPTION
**Pull request by ebb-tide**
Closes #14747

### CODE REVIEW

File: ui/cypress/e2e/dashboardsView.test.ts [hunk 1]
```diff
@@ -53,4 +59,89 @@ describe('Dashboard', () => {
     cy.getByTestID('save-cell--button').click()
     cy.getByTestID('cell--view-empty').should('have.length', 1)
   })
+
+  const getSelectedVariable = (contextID: string, variableID: string) => win =>
+    win.store.getState().variables.values[contextID].values[variableID]
+      .selectedValue
+
+  it('can manage variable state with a lot of pointing and clicking', () => {
```
Comment (drdelambre) [inline ui/cypress/e2e/dashboardsView.test.ts]: :joy:

File: ui/src/index.tsx [hunk 2]
```diff
@@ -125,6 +125,10 @@ const history: History = useRouterHistory(createHistory)({
 export const store = configureStore(loadLocalStorage(), history)
 const {dispatch} = store
 
+if ((window as any).Cypress) {
```
Comment (drdelambre) [inline ui/src/index.tsx]: is there a cypress test setup file we can add this to instead?

File: ui/cypress/e2e/dashboardsView.test.ts [hunk 3]
```diff
@@ -53,4 +59,89 @@ describe('Dashboard', () => {
...
+          cy.getByTestID('variable-dropdown').should('contain', secondKey)
+          cy.window()
+            .pipe(getSelectedVariable(dashboard.id, variable.id))
+            .should('equal', secondKey)
+        })
+      })
+    })
+  })
+
+  it('can create a view through the API', () => {
```
Comment (hoorayimhelping) [inline ui/cypress/e2e/dashboardsView.test.ts]: ❤️

File: ui/cypress/e2e/dashboardsView.test.ts [hunk 4]
```diff
@@ -53,4 +59,89 @@ describe('Dashboard', () => {
     cy.getByTestID('save-cell--button').click()
     cy.getByTestID('cell--view-empty').should('have.length', 1)
   })
+
+  const getSelectedVariable = (contextID: string, variableID: string) => win =>
+    win.store.getState().variables.values[contextID].values[variableID]
+      .selectedValue
+
+  it('can manage variable state with a lot of pointing and clicking', () => {
+    cy.get<Organization>('@org').then(({id: orgID}) => {
```
Comment (hoorayimhelping) [inline ui/cypress/e2e/dashboardsView.test.ts]: @Asalem1 I noticed you changing to
```js
cy.get<Organization>('@org').then(({id: orgID})
``` 
to 
```js
cy.get('@org').then(({id: orgID}: Organization)
```
in some recent PRs - is there a preferred way of doing this?

File: ui/cypress/support/commands.ts [hunk 5]
```diff
@@ -1,5 +1,12 @@
 import {NotificationEndpoint} from '../../src/types'
 
+// const defaultViewQuery = {
+//   text: 'from(bucket: v.mapTypeVar)|> range(start: -15m, stop: now())',
+//   editMode: 'advanced',
+//   builderConfig: {},
+//   name: '',
+// }
```
Comment (hoorayimhelping) [inline ui/cypress/support/commands.ts]: ✂️

File: ui/cypress/e2e/dashboardsView.test.ts [hunk 6]
```diff
@@ -53,4 +59,89 @@ describe('Dashboard', () => {
...
+      cy.createDashboard(orgID).then(({body: dashboard}) => {
+        cy.createMapVariable(orgID).then(({body: variable}) => {
+          cy.fixture('routes').then(({orgs}) => {
+            cy.visit(`${orgs}/${orgID}/dashboards/${dashboard.id}`)
+          })
+          const [firstKey, secondKey] = Object.keys(variable.arguments.values)
```
Comment (hoorayimhelping) [inline ui/cypress/e2e/dashboardsView.test.ts]: i really like destructuring into named variables here! very clean, very readable!

File: ui/cypress/support/commands.ts [hunk 7]
```diff
@@ -53,6 +60,35 @@ export const createCell = (
...
+    })
+  })
+}
+
+export const createDashWithCell = (
+  orgID: string
+): Cypress.Chainable<Cypress.Response> =>
+  createDashboard(orgID).then(({body: dashboard}) => createCell(dashboard.id))
+
+export const createDashWithView = (
```
Comment (hoorayimhelping) [inline ui/cypress/support/commands.ts]: what do you think about changing this function name to indicate that it's also creating a variable? along the lines of `createDashWithViewAndVariable` ?

File: ui/src/timeMachine/actions/index.ts [hunk 8]
```diff
@@ -642,3 +649,20 @@ export const removeCheckThreshold = (level: CheckStatusLevel) => ({
...
+export const loadNewVEO = (fromContextID: string) => (
+  dispatch: Dispatch<Action>,
+  getState: GetState
+): void => {
+  dispatch(
+    setActiveTimeMachine('veo', {view: createView<XYViewProperties>('xy')})
+  )
+
+  const values =
+    get(getState(), `variables.values.${fromContextID}.values`) || {}
```
Comment (hoorayimhelping) [inline ui/src/timeMachine/actions/index.ts]: stylistic suggestion, take it or leave it as you see fit, just putting it out there  :)

[`get`](https://lodash.com/docs/4.17.15#get) lets you specify a default value as its third parameter:

```suggestion
    get(getState(), `variables.values.${fromContextID}.values`, {})
```

File: ui/cypress/e2e/dashboardsView.test.ts (hunk 4 above)
Comment (Asalem1) [reply ui/cypress/e2e/dashboardsView.test.ts]: @hoorayimhelping, @121watts had me do it on my first PR last week and my linter has been yelling at me to change them on files that I visit so I've just been following that. From my perspective it lays out a clearer understanding of what the expectations of the parameter will be by setting it next to the parameter as opposed to the `get`

File: ui/cypress/e2e/dashboardsView.test.ts (hunk 6 above)
Comment (Asalem1) [reply ui/cypress/e2e/dashboardsView.test.ts]: thanks!

File: ui/src/timeMachine/actions/index.ts (hunk 8 above)
Comment (ebb-tide) [reply ui/src/timeMachine/actions/index.ts]: thanks!

**Pull request merged**